### PR TITLE
Make sure to get file model in template picker

### PR DESCRIPTION
**Pull request by julien-nc**
If the template picker creates a file which is not immediately rendered in the file list, it fails to get the FileInfoModel (which is then passed to the default file action). This does not break the Text app because its file action does not use the FileInfoModel. It does break the OnlyOffice app (and maybe others) file creation from a template.

The template picker fails to get the model because `FileList::getModelForFile()` relies on the DOM and the freshly created file is sometimes not added/rendered.

The created file item is not rendered if the target position is out of what's currently visible/loaded.

Fortunately, `FileList::addAndFetchFileInfo()` fetches the file metadata and we can use it to generate the file model in the template picker.
I switched from `await` to `.then()` because I couldn't find another way to get the promise result when multiple parameters are passed to `resolve()`.

We have to be careful when backporting this. So far, I saw that a few changes are necessary to make it work in stable22.

### CODE REVIEW

File: apps/files/src/views/TemplatePicker.vue [hunk 1]
```diff
@@ -213,19 +213,27 @@ export default {
...
-				fileAction.action(fileInfo.basename, {
-					$file: fileList?.findFileEl(this.name),
-					dir: currentDirectory,
-					fileList,
-					fileActions: fileList?.fileActions,
-					fileInfoModel: fileList?.getModelForFile(this.name),
-				})
 
-				this.close()
+				fileList?.addAndFetchFileInfo(this.name).then((status, data) => {
```
Comment (artonge) [inline apps/files/src/views/TemplatePicker.vue:218]: Not pretty, but might simply the code:

```suggestion
                // Use an intermediary `then` to bundle 'status' and 'data' in one destructurable object.
				const { status, data } = await fileList?.addAndFetchFileInfo(this.name).then((status, data) => ({status, data}))
```

**Review by juliusknorr — APPROVED** [review]
Could not trigger it myself but change seems good :+1:

File: apps/files/src/views/TemplatePicker.vue (hunk 1 above)
Comment (eneiluj) [reply apps/files/src/views/TemplatePicker.vue]: @artonge That's exactly what I was looking for. Thank you dear JS ninja :grin:.
Rebased on master. Now uses this shorter and (almost) elegant solution.

**Comment by skjnldsv** [comment]
Lint failures

File: apps/files/src/views/TemplatePicker.vue [hunk 2]
```diff
@@ -213,16 +213,19 @@ export default {
 				)
 				this.logger.debug('Created new file', fileInfo)
 
-				await fileList?.addAndFetchFileInfo(this.name)
+				const { status, data } = await fileList?.addAndFetchFileInfo(this.name).then((status, data) => ({status, data}))
```
Comment (artonge) [inline apps/files/src/views/TemplatePicker.vue:216]: Should fix linting issues:

```suggestion
				const { status, data } = await fileList?.addAndFetchFileInfo(this.name).then((status, data) => ({ data }))
```

**Review by artonge — CHANGES_REQUESTED** [review]

File: apps/files/src/views/TemplatePicker.vue (hunk 2 above)
Comment (eneiluj) [reply apps/files/src/views/TemplatePicker.vue]: We can even go to:
``` javascript
const data = await fileList?.addAndFetchFileInfo(this.name).then((status, data) => data)
```

Rebased and done.

Comment (artonge) [reply apps/files/src/views/TemplatePicker.vue]: Indeed :laughing:

**Review by artonge — APPROVED** [review]

**Comment by eneiluj** [comment]
This can be backported in 21, 22 and 23 IMO.

**Comment by artonge** [comment]
CI failure is unrelated. Merge when you want @eneiluj :)

**Pull request merged**